### ISSUE TITLE
Solution for installing torch, torchvision and torchaudio with Python 3.9

### ISSUE THREAD

**Issue by FengyiLi1102** [hukaixuan19970627/yolov5_obb#571]
I followed the [install.md](https://github.com/hukaixuan19970627/yolov5_obb/blob/master/docs/install.md), and used following codes to create a virtual environment with Python==3.9 and successfully installed corresponding `torch`, `torchvision`, `torchaudio` packages.

The working space is colab.

### Install conda in the colab
```
!pip install -q condacolab
import condacolab
condacolab.install()
```

### Create the virtual environment with Python==3.9
```
!conda create -n env python=3.9 -y
```

### Check your python version
```
%%bash
source activate env
python --version
```
⚠️: You need `%%bash` and `source activate env` for each cell in order to run the following commond within the proper environment.

### Install required `torch`, `torchvision` and `torchaudio` packages.
I found the original downloading page does not contain the required `torch` package, and searched another [one](https://download.pytorch.org/whl/cu113/torch/) to replace. The rest packages are left without any change.
```
%%bash
source activate env
pip3 install torch==1.10.1+cu113 -f https://download.pytorch.org/whl/cu113/torch/
pip3 install torchvision==0.11.2+cu113 torchaudio==0.10.1+cu113 -f https://download.pytorch.org/whl/cu113/torch_stable.html
nvcc -V
```

### Set off and Go
After checking all the version correctness, you can train, detect and predict.

### Tips

- I encountered with a problem of `np.int` for `general.py` and `datasets.py`, and this could be easily solved by just replacing the `np.int` with `np.int32`, which may be caused by the version difference for `numpy` package.
- `%%bash` avoids the verbose training information, and if you want to keep the training information please use `!source activate env; COMMAND1; COMMAND2; ...` to see the log.

**Comment by mimslim** [comment]
Thank you for this solution for working with Python 3.9 to get YOLOV5_OBB working again in the Google Colab environment.   It has mostly worked for me.   My "detect" runs are working now, but my training runs are not.   Seems like a problem with looking at the data files.  (Same data files that were used back with Colab's default was Python 3.7).   I see:
"The DType <class 'numpy._IntegerAbstractDType'> could not be promoted by <class 'numpy.dtypes.StrDType'>."   All my image files are seen as corrupt:  "ignoring corrupt image/label".   Does your Tip about np.int32 vs np.int have anything to do with the integer datatypes error that I am seeing?   And, if so, can you offer a Tip about which lines in general.py and datasets.py need modification?

**Comment by FengyiLi1102** [comment]
> Thank you for this solution for working with Python 3.9 to get YOLOV5_OBB working again in the Google Colab environment. It has mostly worked for me. My "detect" runs are working now, but my training runs are not. Seems like a problem with looking at the data files. (Same data files that were used back with Colab's default was Python 3.7). I see: "The DType <class 'numpy._IntegerAbstractDType'> could not be promoted by <class 'numpy.dtypes.StrDType'>." All my image files are seen as corrupt: "ignoring corrupt image/label". Does your Tip about np.int32 vs np.int have anything to do with the integer datatypes error that I am seeing? And, if so, can you offer a Tip about which lines in general.py and datasets.py need modification?

Hi, for `np.int` in the following https://github.com/hukaixuan19970627/yolov5_obb/blob/b00c3f245e50e7a80460a1b949d22ea7dfeb27a0/utils/general.py#L512
I used `np.int8` instead of `np.int32` and have not encountered any errors that you met.

From you error message 

> "The DType <class 'numpy._IntegerAbstractDType'> could not be promoted by <class 'numpy.dtypes.StrDType'>."

It seems that you were trying to perform some NumPy operations on both int and string type in the same array, and make sure that you lable files follow the requirement of [DOTA format](https://github.com/hukaixuan19970627/yolov5_obb/blob/master/docs/GetStart.md) in the document. I assume that you detection ran correctly because it does not require any label files, right? Therefore, the training proess requiring lable files may expose this error.

**Comment by mimslim** [comment]
I have not added any code to the original YOLOV5_OBB written by Hukaixuan.   I was able to get around the above DType problem by reverting to an older Numpy (1.21.6), with no code changes.   At least the training ran this time around.
